myscript.js

```diff
@@ -141,3 +141,26 @@ form.addEventListener('submit', (e) => {
     form.submit();
   }
 });
+
+const inputs = document.querySelectorAll('.fill-area');
+const dataInput = {
+  fname: '',
+  lname: '',
+  email: '',
+  message: '',
+};
+inputs.forEach((input) => {
+  input.addEventListener('input', () => {
+    dataInput[input.fname] = input.value;
+    dataInput[input.lname] = input.value;
+    dataInput[input.email] = input.value;
+    dataInput[input.message] = input.value;
+    localStorage.setItem('grouped', JSON.stringify(dataInput));
+  });
+});
+const archived = JSON.parse(localStorage.getItem('grouped'));
+if (archived) {
+  inputs.forEach((element) => {
+    element.value = archived[element.name];
+  });
+}
```

Comment:
- [ ] Data is not saving in local storage correctly. After `reloading` the page, the data inside the form is missing. Kindly fix this issue.
  - **Your website:**
   ![ezgif com-gif-maker (5)](https://user-images.githubusercontent.com/60083437/209410960-c57d5133-72c4-4542-9d0e-90901f5cec0c.gif)
  - **Data in Local storage should look like this:**
  ![Screenshot from 2022-12-24 04-38-30](https://user-images.githubusercontent.com/60083437/209410999-d9b8a37c-280f-4e04-a1f3-911cebd41c6d.png)
  - You can consider using `window.onload()` function to get the data from local storage after loading and correctly saving data in local storage. An example is given below:
```
window.onload = () => {
  let savedFormData = localStorage.getItem('contactFormInput');
  savedFormData = JSON.parse(savedFormData);
  // Check if the form data object is found on localStorage
  if (savedFormData) {
  // populate inputs values if data was found
  // ex: nameInput.value = savedFormData.name
  }
}
```